rel/m_lib.c

```diff
@@ -1,371 +1,880 @@
+    }
 
-        stepSize =  (uTarget - uValue) * fraction;
+    stepSize = (uTarget - uValue) * fraction;
 
-        if (stepSize > maxStep) {
-            stepSize = maxStep;
-        } else if (stepSize < minStep) {
-            stepSize = minStep;
-        }
+    if (stepSize > maxStep) {
+      stepSize = maxStep;
+    } else if (stepSize < minStep) {
+      stepSize = minStep;
+    }
 
-        newValue = uValue + (s32) stepSize;
-        if (newValue > uTarget) {
-            newValue = uTarget;
-        }
-        *pValue = newValue;
+    newValue = uValue + (s32)stepSize;
+    if (newValue > uTarget) {
+      newValue = uTarget;
     }
+    *pValue = newValue;
+  }
 
-    return target - *pValue;
+  return target - *pValue;
 }
 
-void rgba_t_move(rgba_t* dest, rgba_t* src) {
-    dest->r = src->r;
-    dest->g = src->g;
-    dest->b = src->b;
-    dest->a = src->a;
+/**
+ * @brief Move the RGBA color value from a source to a destination variable.
+ *
+ * This function copies the RGBA color value from the source variable to the destination variable.
+ *
+ * @param dest Pointer to the destination rgba_t variable.
+ * @param src Pointer to the source rgba_t variable.
+ */
+extern void rgba_t_move(rgba_t* dest, const rgba_t* const src) {
+  dest->r = src->r;
+  dest->g = src->g;
+  dest->b = src->b;
+  dest->a = src->a;
 }
 
-u32 none_proc1(void){
-    return 0;
-    //stub
-} 
-
-void none_proc2(void){
-    //stub
+/**
+ * @brief A function that always returns 0.
+ *
+ * This function returns 0 without performing any operations.
+ *
+ * @return 0
+ */
+extern int none_proc1() { return 0; }
+
+/**
+ * @brief A function that performs no operation.
+ *
+ * This function performs no operation, serving as a placeholder
+ * or no-op function.
+ *
+ * @param actor Pointer to an ACTOR structure.
+ * @param game Pointer to a GAME structure.
+ */
+extern void none_proc2(ACTOR* actor, GAME* game) {}
+
+/**
+ * @brief Check if the game is in pause state.
+ *
+ * This function checks if the game is in the pause state and returns the result.
+ *
+ * @param play Pointer to a GAME_PLAY structure.
+ * @return TRUE if the game is in pause state, FALSE otherwise.
+ */
+extern int _Game_play_isPause(GAME_PLAY* play) { return (play->isPause != 0); }
+
+/**
+ * @brief Calculate a percentage with respect to minimum and maximum values, and apply scaling.
+ *
+ * This function calculates a percentage with respect to given minimum and maximum values.
+ * The percentage is then scaled by the provided scale factor. If shift_by_min is set,
+ * the percentage is shifted by the minimum value.
+ *
+ * @param x Input value to check.
+ * @param min Minimum value for the range.
+ * @param max Maximum value for the range.
+ * @param scale Scaling factor to apply to the percentage.
+ * @param shift_by_min Flag to shift the percentage by the minimum value (1 to shift, 0 not to shift).
+ * @return Scaled percentage of the input value within the specified range.
+ */
```

Comment:
This is what I wrote, might be worth incorporating for clarity's sake
```
/**
 * - If `x` is closer to 0 than `min`, return 0
 * - If `x` is further from 0 than `max`, return the sign of `x`
 * - Otherwise, scale `x` by `scale` and return it.
 * If `shift_by_min`, move `x` by  `min` towards 0 before scaling it.
 */
```